packages/ibm-products/src/components/DataSpreadsheet/DataSpreadsheetBody.tsx

```diff
@@ -628,6 +648,10 @@ export const DataSpreadsheetBody = forwardRef(
         const row = rows?.[index];
         if (rows && rows.length) {
           prepareRow?.(row);
+          let uniqueRowHeaderComponent: any[] | undefined = undefined;
+          if (typeof renderRowHeader === 'function') {
```

Comment:
this check is already defined in `hasCustomRowHeader`. in the render i would just do `{hasCustomRowHeader && renderRowHeader(index}`

Reply:
Fixed thanks! I had to add null check on the function, but was able to get rid of these lines and do singular line. Thanks!